### ISSUE TITLE
UnicodeDecodeError when receiving flag_info info error

### ISSUE THREAD

**Issue by alvarovelezgalvez** [spookylukey/django-paypal#26]
Hi all,

First of all thanks to spookylukey (and also dcramer) for this super library. I have recently discovered that sometimes Paypal is returning some unreadable characters in the _'flag_info'_ field and an UnicodeDecodeException exception is raised in the following line:
_self.flag_info += info_ 
of file: _django_paypal-0.1.3-py2.7.egg/paypal/standard/models.py_

I guess that a try/catch surrounding the line and managing the unicode variable properly should be enough to fix it.

Best regards from Barcelona,
Álvaro Vélez.

**Comment by spookylukey** [comment]
Closing as not enough info was provided, and this is old now.

**Comment by rafaelks** [comment]
Hi,

I'm having this error in my application using version `0.2`. This is the request params that's causing this:

```json
{
  "protection_eligibility": "Ineligible",
  "last_name": "Bar",
  "txn_id": "25F78140PL237475G",
  "receiver_email": "foo@bar.com",
  "payment_status": "Completed",
  "tax": "0.00",
  "residence_country": "BR",
  "invoice": "0c85c58f-df5b-baz-bar-foo",
  "payer_status": "unverified",
  "txn_type": "web_accept",
  "handling_amount": "0.00",
  "payment_date": "17:00:38 Apr 19, 2015 PDT",
  "first_name": "Foo",
  "item_name": "Foobar",
  "charset": "windows-1252",
  "notify_version": "3.8",
  "receiver_id": "27L7EUKFFCSPS",
  "business": "foo@bar.com",
  "payer_id": "W832M3A7B44MU",
  "verify_sign": "AaXhtpwght6PEuy-z7LFdf...",
  "mc_fee": "3.10",
  "mc_currency": "BRL",
  "shipping": "0.00",
  "payer_email": "foobar@gmail.com",
  "payment_type": "instant",
  "mc_gross": "39.00",
  "ipn_track_id": "f17a",
  "quantity": "1"
}
```

**Comment by stefanw** [comment]
I'm seeing this as well. It looks like a Unicode object (`self.flag_info`) is concatenated with a byte object `info` which contains non-ascii characters.

[All assignments of `flag` here](https://github.com/spookylukey/django-paypal/blob/master/paypal/standard/ipn/views.py#L33-L58) are byte objects and the last two may contain non-ascii characters. Not sure if this is the root cause though.

**Comment by rafaelks** [comment]
@spookylukey Can you open this issue again? I'll try to look into it and fix.

**Comment by thiagogds** [comment]
I'm having the same issue.
I looked it up and in fact it is two problems :pensive: 

1st: **UnicodeDecodeError**
The problem and solution proposed by @stefanw is ok. When a form error have no-ascii characters we get the error. It was happening because my django is in pt-BR and my form errors have accents.

The real error in fact was that PayPal `retry_at` field was not in the PayPal date format.
I saw that many fields was already [patched](https://github.com/spookylukey/django-paypal/blob/master/paypal/standard/forms.py#L235). I did a quick test adding `retry_at` in that list and it worked fine. But I didn't run any unit tests yet.

What do you guys think? I can send a PR later.

**Comment by spookylukey** [comment]
The UnicodeDecodeError should have been addressed in https://github.com/spookylukey/django-paypal/commit/686382b58b2a047e2e61906fdc5f2211cef468f1

**Comment by spookylukey** [comment]
Both issues should now be addressed in 0.2.5
